### ISSUE TITLE
helm dep list shows incorrect status for local dependencies

### ISSUE THREAD

**Issue by dhawal55** [helm/helm#2167]
I have a repo structure as follows:

```
├── consul
│   ├── Chart.yaml
│   ├── charts
│   └── values.yaml
├── vault
│   ├── Chart.yaml
│   ├── charts
│   ├── requirements.yaml
│   └── values.yaml
```

My requirements.yaml looks like this:
```
dependencies:
  - name: consul
    version: 0.1.0
    repository: "file://consul"
```

When I run the `helm dep list` command, it shows status 'missing' for local dependencies:
```
helm dep list vault                                                
NAME  	VERSION	REPOSITORY      	STATUS 
consul	0.1.0  	file://../consul	missing
```

**Comment by technosophos** [comment]
@qwangrepos Is this one you'd like to take a look at?

**Comment by rbwsam** [comment]
Looking at [the code](https://github.com/kubernetes/helm/blob/master/cmd/helm/dependency.go#L156) it appears that it only checks the `charts` subdirectory of the chart in question and disregards the `repository` entirely when searching for the dependency.

If we change the code to include the relative path of the given `repository` we will be searching multiple paths and will need to decide on a strategy for handling the case of multiple matching charts. Currently when the code finds multiple versions of a chart in archived format it resolves with a "too many matches" message but does not tell the user what the matches were, making it difficult to resolve the problem.

I want to fix this issue but would greatly prefer to get some input regarding the desired behavior of the code before I get started.

Thanks!

**Comment by dhawal55** [comment]
I haven't been using helm lately so don't know if this is fixed. Can someone test it out?

**Comment by bacongobbler** [comment]
no PRs have addressed this specifically so I imagine this is still in the same state.

**Comment by phemmer** [comment]
Is there any workaround for this issue? I tried doing something like symlinking `../` to the `charts` directory, but helm doesn't seem to like this (CPU spins and it starts gobbling memory).

&nbsp;

This behavior is also highly surprising considering it's documented as working (https://docs.helm.sh/helm/#helm-dependency):

> Starting from 2.2.0, repository can be defined as the path to the directory of the dependency charts stored locally. The path should start with a prefix of “file://“. For example,
```
# requirements.yaml
dependencies:
- name: nginx
  version: "1.2.3"
  repository: "file://../dependency_chart/nginx"
```

**Comment by jeff-knurek** [comment]
I'm trying to look into the original issue, and I either don't know how to reproduce, or I don't know what the actual problem is.

```
helm dep list vault                                                
NAME  	VERSION	REPOSITORY      	STATUS 
consul	0.1.0  	file://../consul	missing
```
the dep will be missing until `helm dep up vault` is executed.

But the thing to note here, the `requirements.yaml` will need to be relative, like what shows in the `dep list` output above. 
```
dependencies:
  - name: consul
    version: 0.1.0
    repository: "file://../consul"
```

------
maybe this is/was resolved since `v2.12` and/or it was a documentation problem
https://v2.helm.sh/docs/developing_charts/#managing-dependencies-with-requirements-yaml
but I think it's been addressed already
